### DISCUSSION TITLE
custom arduino code for spider v1

### DISCUSSION THREAD

**Discussion by nikpoz** [FYSETC/FYSETC-SPIDER#140] [Q&A]
Hi my dear friends!!
I want to use a spider v1 I had laying around to make a small project, not regarding printers.
My issue is I can't find a way to upload my code.
It's my first time dealing with a stm32 mcu, tried all usual ways of uploading marlin firmware but nothing worked so far.
I am out of thoughts why!! so any help would be helpful... 
I am using arduino IDE and my board settings are as follows:

board: Generic STM32F4 series
board part number: Generic STM32F446VETx
U(S)ART support: Enabled (Generic 'Serial')
USB support (if available): None
USB speed (if available): Low/Full speed
Optimize: Smallest; OS Default
Debug symbols and core logs; None
C Runtime library; Newlib Nano(Default)
Upload method; STM32CubeProgrammer DFU

I compile my code as .bin,
I rename it as firmware .bin
and I am uploading using a sd card...

please help me locate my mistakes in this...

**Comment by ejhoness** [comment]
Maybe, the jumper on the boot pin

Em qua., 27 de set. de 2023 às 23:26, nik_poz ***@***.***>
escreveu:

> Hi my dear friends!!
> I want to use a spider v1 I had laying around to make a small project, not
> regarding printers.
> My issue is I can't find a way to upload my code.
> It's my first time dealing with a stm32 mcu, tried all usual ways of
> uploading marlin firmware but nothing worked so far.
> I am out of thoughts why!! so any help would be helpful...
> I am using arduino IDE and my board settings are as follows:
>
> board: Generic STM32F4 series
> board part number: Generic STM32F446VETx
> U(S)ART support: Enabled (Generic 'Serial')
> USB support (if available): None
> USB speed (if available): Low/Full speed
> Optimize: Smallest; OS Default
> Debug symbols and core logs; None
> C Runtime library; Newlib Nano(Default)
> Upload method; STM32CubeProgrammer DFU
>
> I compile my code as .bin,
> I rename it as firmware .bin
> and I am uploading using a sd card...
>
> please help me locate my mistakes in this...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FYSETC/FYSETC-SPIDER/discussions/140>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARM5JY2HR45RU4RBZWZFEC3X4TN4XANCNFSM6AAAAAA5KID6PU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Emerson Jhones
***@***.***
35 88518611

**Reply by nikpoz** [reply]
I can't really remember if I tried it, but I am guessing I didn't. I will check it once more. 
I ended up uploading my code using vscode and platformio. 
Thanks for the reply anyway👍👍👍😊😊😊
